nifi-commons/nifi-xml-processing/src/main/java/org/apache/nifi/xml/processing/transform/StandardTransformProvider.java

```diff
@@ -0,0 +1,118 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.xml.processing.transform;
+
+import org.apache.nifi.xml.processing.ProcessingAttribute;
+import org.apache.nifi.xml.processing.ProcessingException;
+
+import javax.xml.XMLConstants;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import java.util.Objects;
+
+/**
+ * Standard implementation of Transform Provider with secure processing enabled
+ */
+public class StandardTransformProvider implements TransformProvider {
+    private static final boolean SECURE_PROCESSING_ENABLED = true;
+
+    private static final String ENABLED_PROPERTY = "yes";
+
+    private static final String INDENT_AMOUNT_OUTPUT_KEY = "{http://xml.apache.org/xslt}indent-amount";
+
+    private static final String INDENT_AMOUNT = "2";
+
+    private boolean indent;
+
+    private boolean omitXmlDeclaration;
+
+    private String method;
+
+    /**
+     * Set Indent Status
+     *
+     * @param indent Indent Status
+     */
+    public void setIndent(final boolean indent) {
+        this.indent = indent;
+    }
+
+    /**
+     * Set Output Method
+     *
+     * @param method Method or null when default configuration should be used
+     */
+    public void setMethod(final String method) {
+        this.method = method;
+    }
+
+    /**
+     * Set Omit XML Declaration
+     *
+     * @param omitXmlDeclaration Omit XML Declaration
+     */
+    public void setOmitXmlDeclaration(final boolean omitXmlDeclaration) {
+        this.omitXmlDeclaration = omitXmlDeclaration;
+    }
+
+    /**
+     * Transform Source to Result
+     *
+     * @param source Source to be transformed
+     * @param result Result containing transformed information
+     */
+    @Override
+    public void transform(final Source source, final Result result) {
+        Objects.requireNonNull(source, "Source required");
+        Objects.requireNonNull(result, "Result required");
+
+        final TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        final Transformer transformer;
+        try {
+            transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, ProcessingAttribute.ACCESS_EXTERNAL_DTD.getValue());
+            transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, ProcessingAttribute.ACCESS_EXTERNAL_STYLESHEET.getValue());
+            transformerFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, SECURE_PROCESSING_ENABLED);
```

Comment:
could you try these in separate try blocks. xalan doesn't support transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET)? https://github.com/apache/poi/blob/trunk/poi/src/main/java/org/apache/poi/util/XMLHelper.java has similar code to create secure transformers and parsers that has been modified over time as users with old versions of XSL transformers or XML parsers pop up and complain the code fails for them.

Reply:
Thanks for the feedback @pjfanning. Are you referring to old version of Xalan?  The documentation [TransformerFactory.setFeature()](https://docs.oracle.com/javase/8/docs/api/javax/xml/transform/TransformerFactory.html#setAttribute-java.lang.String-java.lang.Object-) documentation for Java 8 indicates the following:

> All implementations that implement JAXP 1.5 or newer are required to support the [XMLConstants.ACCESS_EXTERNAL_DTD](https://docs.oracle.com/javase/8/docs/api/javax/xml/XMLConstants.html#ACCESS_EXTERNAL_DTD) and [XMLConstants.ACCESS_EXTERNAL_STYLESHEET](https://docs.oracle.com/javase/8/docs/api/javax/xml/XMLConstants.html#ACCESS_EXTERNAL_STYLESHEET) properties.

NiFi requires Java 8 as a minimum version, and framework usage of XML transformation does not support overriding the JAXP implementation.

Although separating the calls into different blocks might help clarify what failed, it seems like it would still result in failures on supported JAXP implementations.

Reply:
I've created a JIRA to capture the potential refactor of this module to use something like `trySet` and `quietSet`.  I find that a neat solution to this specific problem.  Thanks for sharing!

For this PR, I think the current solution fits in well with the other work recently done to consolidate XML processing, and doesn't preclude potential improvements in the future.

Reply:
> I've created a JIRA to capture the potential refactor of this module to use something like `trySet` and `quietSet`. I find that a neat solution to this specific problem. Thanks for sharing!

https://issues.apache.org/jira/browse/NIFI-9947